packages/keyring/src/__tests__/Keyring.spec.ts

```diff
@@ -277,3 +278,55 @@ describe('reset', () => {
     expect(Object.keys(localStorage).filter((key) => key.startsWith('account:0x')).length).toEqual(0);
   });
 });
+
+describe('changePassword', () => {
+  const NEW_PASSWORD = 'valid-password';
+
+  it('should throw out error if wallet is not initialized', async () => {
+    const keyring = new Keyring();
+
+    await expect(keyring.changePassword(PASSWORD, NEW_PASSWORD)).rejects.toThrowError(
+      new CoongError(ErrorCode.KeyringNotInitialized),
+    );
+  });
+
+  it('should throw out error if currentPassword is not correct', async () => {
+    const keyring = await initializeNewKeyring();
+
+    await expect(keyring.changePassword('incorrect-password', NEW_PASSWORD)).rejects.toThrowError(
+      new CoongError(ErrorCode.PasswordIncorrect),
+    );
+  });
+
+  it('should initialize the rawMnemonic with new password', async () => {
+    const keyring = await initializeNewKeyring();
+
+    const initializeWallet: Mock = vi.fn();
+    keyring.initialize = initializeWallet;
```

Comment:
We probably could use `vi.spyOn` like the test below instead of reassigning the method here